app/web/features/profile/view/leaveReference/LeaveReferencePage.tsx

```diff
@@ -68,23 +72,27 @@ export default function LeaveReferencePage() {
       )}
       {(isUserLoading || isAvailableReferencesLoading) && <CircularProgress />}
 
-      {availableRefrences &&
+      {availableReferences &&
         user &&
         ((referenceType ===
           referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND] &&
-          availableRefrences.canWriteFriendReference &&
+          availableReferences.canWriteFriendReference &&
           user.friends === User.FriendshipStatus.FRIENDS) ||
         (hostRequestId &&
-          availableRefrences.availableWriteReferencesList.find(
-            ({ hostRequestId: availableId }) => availableId === +hostRequestId
+          availableReferences.availableWriteReferencesList.find(
+            ({ hostRequestId: availableId }) => availableId === hostRequestId
           )) ? (
           <div className={classes.root}>
             <ProfileUserProvider user={user}>
               <Hidden smDown>
                 <UserOverview />
               </Hidden>
               <div className={classes.form}>
-                <ReferenceForm />
+                <ReferenceForm
+                  referenceType={referenceType}
+                  userId={userId}
+                  step={step}
+                />
```

Comment:
```suggestion
                <ReferenceForm
                  hostRequestId={hostRequestId}
                  referenceType={referenceType}
                  userId={userId}
                  step={step}
                />
```
Otherwise host/surfer references are still broken